### PR TITLE
Update ChangeLog for 3.10.0

### PR DESCRIPTION
**Pull request by ann0see**
**Short description of changes**

Changelog for 3.10.0

CHANGELOG: SKIP

**Context: Fixes an issue?**

/
**Does this change need documentation? What needs to be documented and how?**
/

**Status of this Pull Request**
Ready for review

**What is missing until this pull request can be merged?**
Review

## Checklist

-  [x] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/main/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
-  [x] I tested my code and it does what I want
-  [x] My code follows the [style guide](https://github.com/jamulussoftware/jamulus/blob/main/CONTRIBUTING.md#source-code-consistency) 
-  [x] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. 
-  [x] I've filled all the content above

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -1,4 +1,136 @@
-### 3.9.1dev <- NOTE: the release version number will be 3.9.2 ###
+### 3.9.1dev <- NOTE: the release version number will be 3.10.0 ###
+
+- Updated clang version to Ubuntu LTS version (#3020).
```
Comment (ann0see) [inline ChangeLog:3]: I have the feeling that that's not enough somehow. But didn't check it yet.

File: ChangeLog [hunk 2]
```diff
@@ -1,4 +1,136 @@
...
+- Build: Updated bundled Qt6 to version 6.4.2 (#2991).
+  (contributed by @app/github-actions)
+
+- Make checkboxes self-labelling (#2996).
+  (contributed by @pljones)
+
+- Translations updated from Hosted Weblate (#2998).
+  (contributed by @weblate)
+
+- Internal: The JACK build will now install directly from the JACK GitHub repository <!-- Insert a short, end-user understandable sentence in past tense right here, e.g.: Client: Fixed crash when clicking the connect button too fast or SKIP if the change should not be documented in the ChangeLog --> (#3003).
```
Comment (pljones) [inline ChangeLog:84]: Something went wrong here.

File: ChangeLog [hunk 3]
```diff
@@ -1,4 +1,136 @@
...
+- Rename `--directoryserver` to `--directoryaddress` (and internal changes) (#3017).
+  (contributed by @pljones)
+
+- Build: Bump dev-drprasad/delete-tag-and-release from 085c6969f18bad0de1b9f3fe6692a3cd01f64fe5 to 0885ecc4123e071dd1a8be04d3a0578f5390581c (#3027).
+  (contributed by @app/dependabot)
+
+- Build: Updated bundled JACK (Windows-only) to version 1.9.22 (#3031).
+  (contributed by @ann0see)
+
+- Internal: Improved maintainability of Jamulus.pro by using global definitions.<!-- Insert a short, end-user understandable sentence in past tense right here, e.g.: Client: Fixed crash when clicking the connect button too fast or SKIP if the change should not be documented in the ChangeLog --> (#3046).
```
Comment (pljones) [inline ChangeLog:111]: And here.

File: ChangeLog (hunk 2 above)
Comment (ann0see) [reply ChangeLog:84]: Correct. The entry misses a line break causing the html comment to show up

File: ChangeLog [hunk 4]
```diff
@@ -1,5 +1,139 @@
...
+- rename `directoryServer` to `directory` in `jamulusserver/getServerProfile` response (#3048).
+  (contributed by @pljones)
+
+- Modernise / fix some Python and added linting to CI (#3052).
+  (contributed by @declension)
+
+- Save the HTML formatting in Server Welcome Message. (Applies to the GUI Server.) (#3058).
+  (contributed by @pljones)
+
+- Mac: Upgrade legacy build to a new but unsupported SDK. This hopefully still allows the legacy build to exist, but no guarantees are given. You are advised to upgrade your OS (#3066).
```
Comment (pljones) [inline ChangeLog:123]: I think this needs to be shorter and clearer.  Something like "Change minimum support macOS version to X.Y.  Upgrade your OS if you find Jamulus isn't working."

Comment (pljones) [reply ChangeLog:123]: Or something like:
> - Build: Windows 64bit releases use Qt 6.4.3 now. Jamulus will from now on only support Windows 10 and above (#2300).
>  (contributed by @hoffie)

File: ChangeLog [hunk 5]
```diff
@@ -1,5 +1,139 @@
...
+- Autobuild: Hardened detection of QT 5 builds for macOS and iOS (#3057).
+  (contributed by @ann0see)
+
+- Build: Windows 64bit releases use Qt 6.4.3 now. Jamulus will from now on only support Windows 10 and above (#2300).
+  (contributed by @hoffie)
+
+- Translation: Document translation process shift to Weblate. Translators can from now one use Weblate for translations (#2880).
+  (contributed by @ann0see)
+
+- Build: Updated bundled Qt6 to version 6.4.0 (#2892).
```
Comment (pljones) [inline ChangeLog:21]: This needs to say which platform is affected.

File: ChangeLog [hunk 6]
```diff
@@ -1,5 +1,139 @@
...
+- Client: Added SRV-based virtual hosting support (#2930).
+  (contributed by @hoffie)
+
+- Server: ISO country codes can now be supplied in upper or lower case (#2934).
+  (contributed by @Rob-NY)
+
+- Build: Bump maxim-lobanov/setup-xcode from 1.5.0 to 1.5.1 (#2936).
+  (contributed by @app/dependabot)
+
+- Build: Enable signing of macOS binary with self signed certificate. If a code signing certificate is set up but no notarization information given, Autobuild will now sign the builds assuming a self signed certificate is given (#2944).
```
Comment (pljones) [inline ChangeLog:57]: This needs to be shorter.  Just "Enable signing of macOS binary with self signed certificate."?

File: ChangeLog (hunk 5 above)
Comment (pljones) [reply ChangeLog:21]: And eventually merge all the bumps.

File: ChangeLog [hunk 7]
```diff
@@ -1,5 +1,139 @@
...
+- Build: Bump dev-drprasad/delete-tag-and-release from 085c6969f18bad0de1b9f3fe6692a3cd01f64fe5 to 0885ecc4123e071dd1a8be04d3a0578f5390581c (#3027).
+  (contributed by @app/dependabot)
+
+- Build: Updated bundled JACK (Windows-only) to version 1.9.22 (#3031).
+  (contributed by @ann0see)
+
+- Internal: Improved maintainability of Jamulus.pro by using global definitions (#3046).
+  (contributed by @ann0see)
+
+- rename `directoryServer` to `directory` in `jamulusserver/getServerProfile` response (#3048).
```
Comment (pljones) [inline ChangeLog:114]: Maybe I should have added an "RPC:" prefix for this?

File: ChangeLog [hunk 8]
```diff
@@ -1,5 +1,139 @@
...
+  (contributed by @pljones)
+
+- Modernise / fix some Python and added linting to CI (#3052).
```
Comment (pljones) [inline ChangeLog:117]: "Build:" prefix?

File: ChangeLog [hunk 9]
```diff
@@ -1,5 +1,139 @@
...
+- Build: Updated bundled Qt6 to version 6.5.1 (#3070).
+  (contributed by @app/github-actions)
+
+- iOS: Build App with higher SDK version (#3073).
+  (contributed by @ann0see)
+
+- MacOS: Fix building Jamulus via Xcode (#3076).
+  (contributed by @ann0see)
+
+- Removed docs/RELATED-PROJECTS.md from repository. The up to date version is now on jamulus.io (#3091).
```
Comment (pljones) [inline ChangeLog:135]: Mmm... needs _some_ prefix.  Not sure what we use.  "Docs:"?

File: ChangeLog [hunk 10]
```diff
@@ -1,5 +1,139 @@
...
+- Refactor: use `size_t` for vector and array indexes that must not be negative (#2893).
+  (contributed by @pljones)
+
+- Internal: Include HTML in strDirectoryTypeWT to make translation easier (#2894).
+  (contributed by @pljones)
+
+- Android: Fixed a crash on session startup that occurs on many devices (#2905).
+  (contributed by @danryu)
+
+- New command line switch (--jsonrpcbindip) to specify the bind address for the RPC server (#2917).
```
Comment (pljones) [inline ChangeLog:33]: "RPC:" prefix.

File: ChangeLog [hunk 11]
```diff
@@ -1,5 +1,139 @@
...
+  (contributed by @Rob-NY)
+
+- jamulusserver/getClients method expanded to include all elements of a user profile as well as the number of active client connections (#2918).
```
Comment (pljones) [inline ChangeLog:36]: "RPC:" prefix.

File: ChangeLog [hunk 12]
```diff
@@ -1,5 +1,139 @@
...
+- Build: Updated bundled JACK (Windows-only) to version 1.9.21.20221111 (#2960).
+  (contributed by @app/github-actions)
+
+- Build: Updated bundled Qt6 to version 6.4.1 (#2961).
+  (contributed by @app/github-actions)
+
+- Build: Updated bundled Qt6 to version 6.4.2 (#2991).
+  (contributed by @app/github-actions)
+
+- Make checkboxes self-labelling (#2996).
```
Comment (pljones) [inline ChangeLog:75]: "GUI:" prefix.

File: ChangeLog [hunk 13]
```diff
@@ -1,5 +1,139 @@
...
+  (contributed by @pljones)
+
+- Translations updated from Hosted Weblate (#2998).
+  (contributed by @weblate)
+
+- Internal: The JACK build will now install directly from the JACK GitHub repository (#3003).
+  (contributed by @ann0see)
+
+- CONDENSE with https://github.com/jamulussoftware/jamulus/pull/2833 (#3008).
```
Comment (pljones) [inline ChangeLog:84]: Not sure what this means.  Maybe skip change log?

File: ChangeLog [hunk 14]
```diff
@@ -1,5 +1,139 @@
...
+- Internal: Improved Qt Creator 5 compliance of Jamulus.pro (#3010).
+  (contributed by @pljones)
+
+- Linux: Debian users can now use the official PPA to get automatic updates. See the Linux install page on jamulus.io for more information (#3013).
+  (contributed by @ann0see)
+
+- Internal: The JACK build will now install directly from the JACK GitHub repository (#3014).
+  (contributed by @ann0see)
+
+- Rename `--directoryserver` to `--directoryaddress` (and internal changes) (#3017).
```
Comment (pljones) [inline ChangeLog:99]: "Server:" prefix.

File: ChangeLog [hunk 15]
```diff
@@ -1,5 +1,139 @@
...
+  (contributed by @pljones)
+
+- Updated clang version to Ubuntu LTS version (#3020).
```
Comment (pljones) [inline ChangeLog:102]: "Build:" prefix.

File: ChangeLog [hunk 16]
```diff
@@ -1,4 +1,138 @@
...
+
+- CONDENSE with https://github.com/jamulussoftware/jamulus/pull/2833 (#3008).
+  (contributed by @ann0see)
+
+- RPC: Rename `directoryServer` to `directory` in `jamulusserver/getServerProfile` response (#3048).
+  (contributed by @pljones)
+
+- Removed docs/RELATED-PROJECTS.md from repository. The up to date version is now on jamulus.io (#3091).
+  (contributed by @mcfnord)
+- GUI: Make checkboxes self-labelling (#2996).
```
Comment (pljones) [inline ChangeLog:20]: The blank line has vanished.

Comment (ann0see) [reply ChangeLog:20]: Strange. Maybe a bug in the script?

Comment (ann0see) [reply ChangeLog:20]: The content is from a comment, not the PR body. So this might be the problem.

Comment (ann0see) [reply ChangeLog:18]: Fixed. The comment didn't have a new line after the sentence ended.

File: ChangeLog (hunk 4 above)
Comment (ann0see) [reply ChangeLog]: Should be better now.

File: ChangeLog (hunk 5 above)
Comment (ann0see) [reply ChangeLog]: Yes. I believe that's something for close to the release.

File: ChangeLog (hunk 6 above)
Comment (ann0see) [reply ChangeLog]: Should be fixed.

File: ChangeLog (hunk 7 above)
Comment (ann0see) [reply ChangeLog]: Done.

File: ChangeLog (hunk 8 above)
Comment (ann0see) [reply ChangeLog]: Added Autobuild:

File: ChangeLog [hunk 17]
```diff
@@ -1,5 +1,139 @@
...
+- Internal: Improved maintainability of Jamulus.pro by using global definitions (#3046).
+  (contributed by @ann0see)
+
+- rename `directoryServer` to `directory` in `jamulusserver/getServerProfile` response (#3048).
+  (contributed by @pljones)
+
+- Modernise / fix some Python and added linting to CI (#3052).
+  (contributed by @declension)
+
+- Save the HTML formatting in Server Welcome Message. (Applies to the GUI Server.) (#3058).
```
[the remark this answers is not in the document]
Comment (ann0see) [reply ChangeLog]: Done

File: ChangeLog (hunk 10 above)
Comment (ann0see) [reply ChangeLog]: Done.

File: ChangeLog (hunk 11 above)
Comment (ann0see) [reply ChangeLog]: Done

File: ChangeLog (hunk 12 above)
Comment (ann0see) [reply ChangeLog]: Done

File: ChangeLog (hunk 13 above)
Comment (ann0see) [reply ChangeLog]: Again, something for the last release. I think maybe we could automate this too. This should be added to another changelog entry.

File: ChangeLog (hunk 14 above)
Comment (ann0see) [reply ChangeLog]: Done

**Comment by ann0see** [comment]
Ok. Updated some stuff of the changelog. The translation part needs updating, but that needs to happen close to the release anyway.

**Review by pljones — APPROVED** [review]

**Pull request merged**